mteb/tasks/ZeroshotClassification/eng/RenderedSST2.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks import AbsTaskZeroshotClassification
+
+
+class RenderedSST2(AbsTaskZeroshotClassification):
+    metadata = TaskMetadata(
+        name="RenderedSST2",
+        description="RenderedSST2.",
+        reference="https://huggingface.co/datasets/clip-benchmark/wds_renderedsst2/commits/main",
+        dataset={
+            "path": "clip-benchmark/wds_renderedsst2",
+            "revision": "66b9a461eda025201dd147e5f390f5984c33643a",
+        },
+        type="ZeroShotClassification",
+        category="i2t",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=("2016-01-01", "2016-12-31"),
+        form=["written"],
+        domains=["Movie"],
+        task_subtypes=[],
+        license="mit",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="""d""",
+        n_samples={"test": 1820},
+        avg_character_length={"test": 10.0},
```

Comment:
So here is my two cents on the metadata generalizaton for images:

```suggestion
    metadata = TaskMetadata(
        name="RenderedSST2", # works
        description="RenderedSST2.", # works
        reference="https://huggingface.co/datasets/clip-benchmark/wds_renderedsst2/commits/main", # works
        dataset={ 
            "path": "clip-benchmark/wds_renderedsst2",
            "revision": "66b9a461eda025201dd147e5f390f5984c33643a",
        },
        type="ZeroShotClassification", # works
        category="i2t", # works
        eval_splits=["test"], # works
        eval_langs=["eng-Latn"], # still works (but can be empty for images dataset)
        main_score="accuracy", # works
        date=("2016-01-01", "2016-12-31"), # works
        form=["written"], # this is close to modality
        domains=["Movie"], # works but see ** below
        task_subtypes=["Rendered Texts Understanding"], # works
        license="mit", # work
        socioeconomic_status="mixed", # have considered deleting this globally (probably a good idea to do it now)
        annotations_creators="human-annotated", # works
        dialect=[], # doesn't make a lot of sense
        text_creation="created",
        bibtex_citation="""d""", # works
        n_samples={"test": 1820}, # see * below
        avg_character_length={"test": 10.0},
```


* We could convert these to computed metadata - e.g. computed as a part of #1049. Depending on the dataset it can contain different things. In that case, n_samples and avg_character_length become something like descriptive_statistics: dict[str, Any]

** Is Movie a domain?

Lastly let us fill these up from the start (lot of regrets from MMTEB that I didn't do that - poeple build on examples and if examples are bad the patterns are reproduced.

Reply:
Actually, form can probably be merged into domains

Reply:
I actually have a newer version of this here: #1054, where I call it `sample_creation`. This change works across image and text (and future modalities such as audio files)

Reply:
Great ideas! Will follow these when adding datasets! yeah removing `socioeconomic_status` and change `text_creation` to `sample_creation` works. What do we have in mind for naming when merging forms into domains? @KennethEnevoldsen 

And yeah languages are still relevant and I agree this will add values to the benchmark for multiliungual image-text retrieval! @isaac-chung

Reply:
> What do we have in mind for naming when merging forms into domains? @KennethEnevoldsen

Would just add "Written" and "Transcribed/Subtitles" (since "Spoken" could sound like it was audio) to domain and then remove "form".

Then add a validation for all modality="text" tasks that they should have one or the other.

Reply:
that makes sense. Should we add a modality field? For domain, I will add suitable options for images as well (such as photograghed, rendered, screenshot, generated, etc), parallel to written and transcribed in texts.

Reply:
Yes I think so (see also #1054)